sriovnet_switchdev.go

```
@@ -226,6 +226,44 @@ func findNetdevWithPortNameCriteria(criteria func(string) bool) (string, error)
 	return "", fmt.Errorf("no representor matched criteria")
 }
 
+// Finds the VF index of a port, if applicable
+func GetVfIndexFromNetDevice(netdev string) (int, error) {
+	if !isSwitchdev(netdev) {
+		return 0, fmt.Errorf("netdev %s is not an eswitch port", netdev)
+	}
+
+	physPortNameStr, err := getNetDevPhysPortName(netdev)
+	if err != nil {
+		return 0, fmt.Errorf("netdev %s has no physical backend", netdev)
+	}
+
+	_, vfRepIndex, err := parsePortName(physPortNameStr)
+	if err != nil {
+		return 0, fmt.Errorf("netdev %s is not a virtual function", netdev)
+	}
+
+	return vfRepIndex, nil
+}
+
+// Finds the SF index of a port, if applicable
+func GetSfIndexFromNetDevice(netdev string) (int, error) {
+	if !isSwitchdev(netdev) {
+		return 0, fmt.Errorf("netdev %s is not an eswitch port", netdev)
+	}
+
+	physPortNameStr, err := getNetDevPhysPortName(netdev)
+	if err != nil {
+		return 0, fmt.Errorf("netdev %s has no physical backend", netdev)
+	}
+
+	sfRepIndex, err := sfIndexFromPortName(physPortNameStr)
+	if err != nil {
+		return 0, fmt.Errorf("netdev %s is not a scalable function", netdev)
```

no need to hide underlying error.

`return 0, fmt.Errof("failed to get SF index from physical port name. %s %v", physPortNameStr, err)`

---

Ack